src/foam/nanos/cron/CronScheduler.java

```diff
@@ -32,13 +32,20 @@
    * @return Date of the minimum scheduled cron job
    */
   private Date getMinScheduledTime() {
-    Min min = (Min) MLang.MIN(Cron.SCHEDULED_TIME);
-    cronDAO_.select(min);
-    return ((Cron) min.getValue()).getScheduledTime();
+    Min min = (Min) cronDAO_.select(
+      MLang.MIN(Cron.SCHEDULED_TIME)
+    );
+
+    if ((int) min.getValue() == 0) {
```

Comment:
Here you're casting to an (int) but below you're casting to a (Date).

Reply:
Thanks. Pushing a fix.